### DISCUSSION TITLE
[Feature Request]: Rename PP Solution adhoc

### DISCUSSION THREAD

**Discussion by jonaswre** [microsoft/AL-Go#1563] [Ideas]
I've started using the new PPPreview and i realised one cool thing that would make this even more useful. 

You can use the PPPreview to deploy a solution for each company of a customer. 
Currently I'm only able to deploy the solution to a new enviroment for each company. Which is okay but comes with cost and management overhead. I think it would be useful to rename the solution inflight. This would allow you to deploy multiple solutions into the same enviroment. 

I've had a quick look ad it seem like this would fairly simple to implement.

**Comment by freddydk** [comment]
@andersgMSFT - for comments

**Comment by andersgMSFT** [comment]
Hi Jonas 

Thanks for the feedback and sorry for the slow response!

I am not sure I understand the suggestion. So, I wanted to ask clarifying questions to make sure I have the right idea. 

We did look at the capability for our build tools to prefix the solutions with a tag (like debug/dev) etc. to allow multiple solutions to be deployed to the same environment, but we ran into some problems with the contend of the solution (id vs name), so we ended up abandoning that idea. 

Is the scenario you’re looking for something like that or more along the deployment tools supporting multiple solutions in one repository?

**Comment by jonaswre** [comment]
Hi, no worries. 

Yes I would like to prefix/postfix the solution name. But I wouldn't use dev or debug, I would use the company id. 

There is currently the issue - If you have 10 companies you need 10 Environments with each 1 solution. I would rather have 1 environment and 10 solutions. Thats more cost effective. 

I think this is a common pattern. If you want to use power apps to quickly deploy custom apps.

**Comment by andersgMSFT** [comment]
Okay, I think I understand.

Just as a note if we prefix/postfix the solution, we will need to prefix/postfix all the items (Apps. flows etc.) in the solution as well - since the different solutions would just reference (and overwrite on import) the same items. We did try to prototype this approach early on, but the name and id's of the apps caused some issues, so we ended up abandoning it. 

But it might be worth re-visiting again. 

An alternative approach to solving the issue you describe could be to have multi company support build into the app. 

We are looking at the Business Central connection more flexible, but at the moment, it is only supported if you use flows as a proxy to get and update Business Central data - which is not ideal. 

Would that be a solution to the problem you are describing, or would you prefer the prefix/postfix approach?
(I can't guarantee that we'll implement it, but I can add you input to the proposed backlog, and we'll review it and get back to you :))

**Comment by jonaswre** [comment]
Yes making the BC Connector more flexible, would be the preferred solution. I don't really want the prefixing.
